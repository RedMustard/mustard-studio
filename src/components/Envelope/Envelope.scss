@import '../../style/variables.scss';

.envelope {
    border: 1px solid #fff;
    color: #fff;
    max-width: 450px;
    min-height: 135px;
    min-width: 150px;
    padding: 10px;

    &__header {
        align-items: center;
        display: flex;
        justify-content: space-between;
        margin-bottom: 15px;
    }

    &__modifiers {
        align-items: flex-start;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
    }
}
