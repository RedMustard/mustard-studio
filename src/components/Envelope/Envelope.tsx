import { h } from 'preact';
import { Attack } from './Attack/Attack';
import { Decay } from './Decay/Decay';
import { Release } from './Release/Release';
import { Sustain } from './Sustain/Sustain';


export const Envelope = () => (
    <div className="envelope">
        <div className="envelope__header">
            <div className="envelope__label">
                Envelope
            </div>
        </div>

        <div className="envelope__modifiers">
            <Attack />
            <Decay />
            <Sustain />
            <Release />
        </div>
    </div>
);
