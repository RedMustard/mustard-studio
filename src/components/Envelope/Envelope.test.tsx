import { h } from 'preact';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Envelope } from './Envelope';


describe('<Envelope />', () => {
    it('renders with basic props', () => {
        const wrapper = shallow(
            <Envelope />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
