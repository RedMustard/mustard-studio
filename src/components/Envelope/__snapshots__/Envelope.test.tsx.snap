// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Envelope /> renders with basic props 1`] = `
<div
  className="envelope"
>
  <div
    className="envelope__header"
  >
    <div
      className="envelope__label"
    >
      Envelope
    </div>
  </div>
  <div
    className="envelope__graph"
  />
  <div
    className="envelope__modifiers"
  >
    <Attack />
    <Decay />
    <Sustain />
    <Release />
  </div>
</div>
`;
